lib/index.ts

```diff
@@ -131,6 +130,44 @@ class LogService {
     }
   }
 
+  addLogger(
+    loggerScopeFunction: (config: InitialConfig) => LoggerPlugin,
+    pluginConfig: PluginConfig) {
+    if (typeof loggerScopeFunction !== 'function')
+      return;
+
+    const loggerOptions = loggerScopeFunction.call(this, this.globalConfig);
+
+    if (loggerOptions == null)
+      return;
+
+    const loggerKeys = Object.keys(loggerOptions);
+    const requiredKeys = ['error', 'warn', 'info', 'debug'];
+    const isValidObject = requiredKeys
+      .every((method) => loggerKeys.indexOf(method) !== -1)
+
+    if (!isValidObject)
+      throw new Error(
+        `Invalid Logger, missing one or more of required keys: [${requiredKeys.join(', ')}]`
+      );
+
+    const { silent, level, prettify } = pluginConfig;
+
+    const loggerFormat = combine(
+      label({ label: process.pid.toString() }),
+      timestamp(),
+      prettify ? prettyPrint({ colorize: true, depth: 4 }) : this.simplifyPrint()
+    );
+
+    this.loggers.push({
+      ...pluginConfig,
+      ...loggerOptions,
+      silent: silent ?? false,
+      level: level ?? LogLevel.debug,
+      format: loggerFormat,
+    });
+  }
```

Comment:
format is not being applied, I think you can explore the method `transform`in your loggerFormat, like: `loggerFormat.transform({message: 'incoming msg', level: 'incoming Level Or debug'})` and make the return of it being passed to log methods (`info`, `debug`, `warn`, `error`) and make this format feature as disableable

Reply:
Great, I'm going to remove it

Reply:
I wasnt talking about to remove it